lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java

```diff
@@ -63,20 +64,21 @@ public LockImmutableTimestampResponse lockImmutableTimestamp() {
                 LeasedLockToken.of(response.immutableTimestamp().getLock(), response.getLease()));
     }
 
-    public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        StartAtlasDbTransactionResponseV3 response = delegate.deprecatedStartTransaction(
-                StartIdentifiedAtlasDbTransactionRequest.createForRequestor(clientId));
+    StartTransactionResponseV4 startTransactions(int batchSize) {
+        StartTransactionRequestV4 request = StartTransactionRequestV4.createForRequestor(clientId, batchSize);
+        StartTransactionResponseV4 response = delegate.startTransactions(request);
 
-        Lease lease = response.getLease();
+        Lease lease = response.lease();
```

Comment:
still confused between lease and lock token share and why they're separate, but that can wait

Reply:
I suppose my confusion comes from the fact that the response encodes both `LockLease` and `LockTokenShare` data, yet we want to establish an abstraction/barrier between the two. However, it's already leaked/it's already broken.

Reply:
did you figure this out? We can quickly chat through it if you'd like.